Tests/IBMSwiftSDKCoreTests/SDKInitializerTests.swift

```diff
@@ -0,0 +1,222 @@
+/**
+* Copyright IBM Corporation 2020
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+**/
+
+import Foundation
+
+import XCTest
+@testable import IBMSwiftSDKCore
+
+class SDKInitializerTests: XCTestCase {
+
+    #if os(Linux)
+    static var allTests = [
+        ("testMySDKInit", testMySDKInit),
+        ("testMySDKNewInstance", testMySDKNewInstance),
+        ("testMySDKNewInstanceWithAuthenticator", testMySDKNewInstanceWithAuthenticator),
+        ("testMySDKNewInstanceWithServiceName", testMySDKNewInstanceWithServiceName),
+        ("testYourSDKInit", testYourSDKInit),
+        ("testYourSDKInitWithAuthenticator", testYourSDKInitWithAuthenticator),
+        ("testYourSDKInitWithServiceName", testYourSDKInitWithServiceName),
+    ]
+    #endif
+
+    // MARK: SDK Initializer methods
+
+    public class Shared {
+        public static let userAgent = "this is the user agent string"
+    }
+
+    // MySDK is the code generated for a service that does not specify `includeExternalConfig = true`
+    public class MySDK {
```

Comment:
Overall this looks good, but one suggestion would be to give these classes more contextual meaning. Instead of `MySDK` and `YourSDK` with comments defining what that means, we could rename these to `ExternallyConfigurableSDK` and `NotExternallyConfigurableSDK`. Those class names are long, but I don't mind long names if they self-document.

Reply:
The comment just above the SDK describes the scenario precisely, so I don't think we need to restate all that in the SDK name, but I concede that "My" and "Your" might be misconstrued so I'll pick some other names.